onnx/test/cpp/parser_test.cc

```diff
@@ -238,6 +238,24 @@ TEST(ParserTest, OptionalInputTest) {
   EXPECT_EQ(n.input(2), "z");
 }
 
+TEST(ParserTest, LeadingOptionalInputTest) {
```

Comment:
I assume the other scenario mentioned in parser.h are already tested.

Reply:
Indirectly. But I will add explicit test-cases to cover all, that will be better